packages/devtools_app/lib/src/memory/flutter/memory_heatmap.dart

```diff
@@ -10,37 +10,160 @@ import 'package:flutter/material.dart' hide TextStyle;
-  });
+    this.controller,
+  );
 
-  final InstructionsSize sizes;
-  final Color lightColor;
-  final Color darkColor;
+  final MemoryController controller;
+
+  @override
+  FlameChartState createState() => FlameChartState(controller);
+}
+
+class FlameChartState extends State<FlameChart> with AutoDisposeMixin {
+  FlameChartState(this.controller);
+
+  InstructionsSize sizes;
+
+  Map<String, Function> callbacks = {};
+
+  /// Flame chart has a blueish color.
+  Color lightColor = const Color(0xFFBBDEFB);
+
+  Color darkColor = const Color(0xFF0D47A1);
+
+  MemoryController controller;
+
+  Widget snapshotDisplay;
+
+  _FlameChart _flameChart;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    controller = Provider.of<MemoryController>(context);
+
+    cancel();
+
+    addAutoDisposeListener(controller.selectedSnapshotNotifier, () {
+      setState(() {
+        controller.computeAllLibraries(true, true);
+
+        sizes = InstructionsSize.fromSnapshop(controller);
+      });
+    });
+
+    addAutoDisposeListener(controller.filterNotifier, () {
+      setState(() {
+        controller.computeAllLibraries(true, true);
+      });
+    });
+
+    addAutoDisposeListener(controller.searchNotifier, () {
+      setState(() {
+        final searchingValue = controller.search;
+        if (searchingValue.isNotEmpty) {
+          final Node node = findNode(searchingValue);
+          if (node != null) {
+            selectNode(node);
+          }
+        }
+      });
+    });
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return _FlameChart(sizes, lightColor, darkColor);
+    sizes = InstructionsSize.fromSnapshop(controller);
+
+    if (sizes != null) {
+      _flameChart = _FlameChart(sizes, lightColor, darkColor, callbacks);
+      return _flameChart;
+    } else {
+      return const Text('');
+    }
+  }
+
+  Node findNode(String searchValue) {
+    final FindNodeFunction callback = _flameChart.callbacks[findNodeKey];
+    return callback(searchValue);
+  }
+
+  void selectNode(Node nodeValue) {
+    final SelectNodeFunction callback = _flameChart.callbacks[selectNodeKey];
+    callback(nodeValue);
   }
 }
 
+/// Key in callbacks map.
+const String findNodeKey = 'findNode';
+
+/// findNode callback signature.
+typedef FindNodeFunction = Node Function(String);
+
+/// Key in callbacks map.
+const String selectNodeKey = 'selectNode';
+
+/// selectNode callback signature.
+typedef SelectNodeFunction = void Function(Node);
+
 class _FlameChart extends LeafRenderObjectWidget {
-  const _FlameChart(this.sizes, this.lightColor, this.darkColor);
+  const _FlameChart(
+      this.sizes, this.lightColor, this.darkColor, this.callbacks);
 
   final InstructionsSize sizes;
+
   final Color lightColor;
+
   final Color darkColor;
 
+  final Map<String, Function> callbacks;
```

Comment:
it would be more consistent with Flutter style to enumerate each callback with a typed Function rather than having a map of string names to callbacks.